presto-spi/src/main/java/com/facebook/presto/spi/block/IntArrayList.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi.block;
+
+import java.util.Arrays;
+
+/**
+ * A simplified version of fastutils IntArrayList for the purpose of positions copying.
+ */
+class IntArrayList
+{
+    private static final int MAX_ARRAY_SIZE = Integer.MAX_VALUE - 8;
+
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private int[] array;
+    private int size;
+
+    public IntArrayList(int initialCapacity)
+    {
+        if (initialCapacity < 0) {
+            throw new IllegalArgumentException("Initial capacity (" + initialCapacity + ") is negative");
+        }
+        array = new int[initialCapacity];
+    }
+
+    public IntArrayList()
+    {
+        this(DEFAULT_INITIAL_CAPACITY);
+    }
+
+    public int[] elements()
+    {
+        return array;
+    }
+
+    private void grow(int newCapacity)
+    {
+        if (newCapacity > array.length) {
+            int newLength = (int) Math.min(Math.max(2L * (long) array.length, (long) newCapacity), MAX_ARRAY_SIZE);
+            array = Arrays.copyOf(array, newLength);
+        }
+    }
+
+    public void add(int element)
+    {
+        grow(size + 1);
+        array[size++] = element;
+    }
+
+    public int size()
+    {
+        return size;
+    }
+
+    public boolean isEmpty()
+    {
+        return size == 0;
+    }
+}
```

Comment:
equals, hashcode, tostring?

Reply:
Is that really required since it won't be used? It would essentially be a dead code apart from testing (e.g: `SelectedPositions` doesn't contain equals, hashcode, tostring either). Since this class is package private it's use cases are limited. Maybe add methods when sb uses it?